### PR TITLE
Allow manual input of vaccination rate

### PR DESCRIPTION
**Pull request by oxytocinlove**
If we don't have vaccination data for a location, allow user to manually input a vaccination rate.

### CODE REVIEW

File: src/data/calculate.ts [hunk 1]
```diff
@@ -299,22 +299,28 @@ export const calculatePersonRiskEach = (
...
+      data.percentFullyVaccinated === null ||
       data.averageFullyVaccinatedMultiplier === null
     ) {
       return unadjustedRisk
     }
+    const unvaccinatedPrevalenceRatio =
+      data.unvaccinatedPrevalenceRatio !== null
+        ? data.unvaccinatedPrevalenceRatio
+        : 1 /
+          (data.averageFullyVaccinatedMultiplier * data.percentFullyVaccinated +
```
[the remark this answers is not in the document]
Comment (oxytocinlove) [reply src/data/calculate.ts:311]: I'm pretty sure we actually do pass percentages around, `casesIncreasingPercentage` and `positiveCasePercentage` appear to be percentages on my machine. These also make more sense for user-input data.

[the remark this answers is not in the document]
Comment (beshaya) [reply src/data/calculate.ts:311]: :'(

Strong agree that it makes sense for users to enter percentages, but you could convert to decimal at the input.

If you want to keep it as percentages, though, don't you need to do `data.percentFullyVaccinated / 100` to convert to fraction here?

[the remark this answers is not in the document]
Comment (oxytocinlove) [reply src/data/calculate.ts]: If you convert to decimal at the input you get floating point issues. 

And yeah, that's fixed in the latest commit.

File: src/data/__tests__/calculate.test.ts [hunk 2]
```diff
@@ -544,4 +545,86 @@ describe('calculate', () => {
...
+      unvaccinatedPrevalenceRatio: null,
+    }
+    const defaultUnvaccinatedPrevalenceRatio = 1 / (0.4 * 0.4 + 0.6)
+
+    it.each`
+      theirVaccine
+      ${'vaccinated'}
+      ${'unvaccinated'}
+    `(
+      'Should decrease risk for lower vaccination rate when interacting with $theirVaccine people',
```
Comment (beshaya) [inline src/data/__tests__/calculate.test.ts:563]: Huh, this is right but feels counterintuitive to me.

File: src/data/__tests__/calculate.test.ts [hunk 3]
```diff
@@ -544,4 +545,86 @@ describe('calculate', () => {
...
+        )
+      },
+    )
+
+    it.each`
+      theirVaccine
+      ${'vaccinated'}
+      ${'unvaccinated'}
+    `(
+      'Should increase risk for higher vaccination rate when interacting with $theirVaccine people',
```
Comment (beshaya) [inline src/data/__tests__/calculate.test.ts:585]: Personally I would have done this as a multi-column it.each, but I don't insist.

**Review by beshaya — APPROVED** [review]

File: src/data/__tests__/calculate.test.ts (hunk 2 above)
Comment (oxytocinlove) [reply src/data/__tests__/calculate.test.ts:563]: Yeah, it was counterintuitive to me too. I think the right way to think about it is that holding all the other data (case rates etc) static, a lower vaccination rate means that you have that same case rate with less vaccination which means the chance of getting COVID is lower (and vice versa of course)

**Pull request merged**